lib/vdc/DCRAM.cpp

```diff
@@ -0,0 +1,592 @@
+    _dimsMap["curveDimZ"] = DC::Dimension("curveDimZ", 8);
+    
+    // Coords
+    // ==================
+    _coordVarsMap["curveCoordX"] = CoordVar("curveCoordX", "m", DC::FLOAT, periodic, /*axis=x*/0, /*uniformHint=*/false, {"curveDimX", "curveDimY"}, /*timeDim*/"");
+    _coordVarsMap["curveCoordY"] = CoordVar("curveCoordY", "m", DC::FLOAT, periodic, /*axis=y*/1, /*uniformHint=*/false, {"curveDimX", "curveDimY"}, /*timeDim*/"");
+    _coordVarsMap["curveCoordZ"] = CoordVar("curveCoordZ", "m", DC::FLOAT, periodic, /*axis=z*/2, /*uniformHint=*/false, {"curveDimX", "curveDimY", "curveDimZ"}, /*timeDim*/"");
+
+    // Mesh
+    // ==================
+    vector<string> curveDims = {"curveDimX", "curveDimY", "curveDimZ"};
+    vector<string> curveCoords = {"curveCoordX", "curveCoordY", "curveCoordZ"};
+    DC::Mesh curveMesh("curveMesh", curveDims, curveCoords);
+    _meshMap[curveMesh.GetName()] = curveMesh;
+    
+    _dataVarsMap["curveData"] = DC::DataVar("curveData", "", DC::FLOAT, periodic, curveMesh.GetName(), /*timeCoordVar*/"", DC::Mesh::NODE);
+#endif
+}
+
+
+void DCRAM::AddDimension(const DC::Dimension &dim)
+{
+//    assert(_dimsMap.find(dim.GetName()) == _dimsMap.end());
+    _dimsMap[dim.GetName()] = dim;
+}
+
+
+void DCRAM::AddMesh(const DC::Mesh &mesh)
+{
+//    assert(_meshMap.find(mesh.GetName()) == _meshMap.end());
+    _meshMap[mesh.GetName()] = mesh;
+}
+
+
+void DCRAM::AddCoordVar(const DC::CoordVar &var, const float *buf)
+{
+//    assert(_coordVarsMap.find(var.GetName()) == _coordVarsMap.end());
+    _coordVarsMap[var.GetName()] = var;
+    
+    size_t size = 1;
+    auto dimNames = var.GetDimNames();
+    for (auto name : dimNames) {
+        DC::Dimension dim;
+        getDimension(name, dim);
+        size *= dim.GetLength();
+    }
+    copyVarData(var, buf, size);
+}
+
+
+void DCRAM::AddDataVar(const DC::DataVar &var, const float *buf)
+{
+//    assert(_dataVarsMap.find(var.GetName()) == _dataVarsMap.end());
+    _dataVarsMap[var.GetName()] = var;
+    
+    size_t size = 1;
+    vector<size_t> dimLens;
+    GetMeshDimLens(var.GetMeshName(), dimLens);
+    for (auto len : dimLens)
+        size *= len;
+    copyVarData(var, buf, size);
+}
+
+
+void DCRAM::copyVarData(const DC::BaseVar &var, const float *buf, const size_t size)
+{
+    if (_dataMap.count(var.GetName()))
+        delete [] _dataMap[var.GetName()];
+    
+    float *copy = new float[size];
+    memcpy(copy, buf, sizeof(float)*size);
+    _dataMap[var.GetName()] = copy;
+}
+
+
+bool DCRAM::getDimension(string dimname, DC::Dimension &dimension) const
+{
+    return getDimension(dimname, dimension, -1);
+}
+
+bool DCRAM::getDimension(string dimname, DC::Dimension &dimension, long ts) const
+{
+    map<string, DC::Dimension>::const_iterator itr;
+    itr = _dimsMap.find(dimname);
+    if (itr == _dimsMap.end()) return (false);
+    dimension = itr->second;
+
+    if (dimension.IsTimeVarying()) {
+        long len = _ncdfc->GetDimLengthAtTime(dimname, ts);
```

Comment:
Here and elsewhere the private member variable `_ncdfc` is dereferenced, but I don't ever see where it is assigned.

Reply:
Good catch, thanks. I had removed most of the uses of `_ncdfc` and the ones I hadn't should theoretically never be reached which is why this didn't result in any errors. I removed the remaining references to `_ncdfc`.